addons/website/models/website_page.py

```diff
@@ -43,6 +43,33 @@ class Page(models.Model):
     website_id = fields.Many2one(related='view_id.website_id', store=True, readonly=False, ondelete='cascade')
     arch = fields.Text(related='view_id.arch', readonly=False, depends_context=('website_id',))
 
+    # Bindings to model/records, to expose the page on the website.
+    # Route: /model/<string:page_name_slugified
+    page_name = fields.Char(string="Page Name", help="The name of the page")
+    name_slugified = fields.Char(compute="_compute_page_url", store=True,
+        string="URL Name", help="The name of the page usable in a url")
+    page_type = fields.Selection(selection=[("listing", "Listing"), ("single", "Single record")],
+        string="Page Type", help="The type of the page. If set, it indicates whether the page displays a list of records or a single record")
+    record_domain = fields.Char(string="Domain", help="Domain to restrict records that can be viewed publicly")
```

Comment:
Couldn't this be done by adding `ir.rule` containing domain force, through studio? Like if you create a custom model and want to interface it on the website, instead of adding a domain here it would just create those IR RULES so that the model is "generically" publicly restricted?
And that would avoid yet another field on website.page but would use standard odoo?

Reply:
It could in theory, we decided not to for the following reasons:
- ir.rules are not surfaced in Studio (only ir.model.access is), so the amount of work would be quite greater and Studio has historically not been involved at all with ir.rules (with the notable exception of multi-company stuff), we didn't want to change that now
- it made sense to group the domain for visibility there functionally ('here are the records i want to see on my special page that i'm configuring right now')